### ISSUE TITLE
Performance drop compared with original SECOND implementation?

### ISSUE THREAD

**Issue by JingwenWang95** [open-mmlab/mmdetection3d#1061]
Hi! Thanks for your great work, it is really a nice codebase! I used original SECOND implementation as 3D detector for my SLAM project, and I just wanted to switch to your implementation as this is really an awsome project. However, when I run your inference code the results seem to degrade much compared with the original SECOND implementation. I'd like to seek some help on how to solve this.

Here is the details of the problem: I use SECOND as 3D LiDAR detector in a SLAM system running on KITTI dataset. I also need some detections behind, so I set the pointcloud range (x-dim) to [-20, 50.4] (I'm not sure if I did it correctly but I changed everything in the config file from [0, 70.4] to [-20, 50.4]). I did the same thing to original SECOND as well. But as I compare the results, it seems your implementation missed some detcetions, and the detected boxes are also slightly worse. Here is the comparison:

![000115-original](https://user-images.githubusercontent.com/19255811/142450133-0425117f-5bfc-4878-be55-832cf394d59e.png)
![Screenshot from 2021-11-18 15-58-06](https://user-images.githubusercontent.com/19255811/142450552-69aa9245-dc30-449e-9824-c391dd02bff9.png)

The top image shows the result from original SECOND and the bottom is your implementation, both run on frame 115 of KITTI-07. I don't know why this is the case. I'm not sure if it's because I have done something wrong with the inference code or something else. I'd be appreciated if you can provide some help or insight with this!

Thank you in advance!

**Comment by Tai-Wang** [comment]
So it seems that both are able to detect objects behind but ours seem a little worse? It's hard to tell the key difference between them from only one frame. Have you ever evaluated them quantitatively on the whole validation set?

Actually, we support many other datasets involving 360-degree point clouds. You can also try with them if convenient.

**Comment by JingwenWang95** [comment]
Hi, thanks for your reply! Yes, it seems your version is slightly worse. As I'm running SLAM with it I did not test quantitatively on the validation set, instead I run it on an entire sequence of KITTI odometry dataset. And the SLAM results I got are much worse: many objects got reconstructed but disappeared soon due to lack of repeated detection. That's why I visualized the results on individual frames trying to figure out what happened. Actually the result shown here is just an example, I tested on several frames and the results are similar: Your version tends to miss some detections, especially when the object's x-coordinate is close to 0.

Actually, I resorted to having a try with your PointPillars, and it works much better. I got comparable SLAM results with using original SECOND. So maybe I can switch to PointPillars. But I'm still not sure what happened with SECOND.
